src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+     */
+    @DefaultSerializer(QNameAndInterval.Serializer.class)
+    @VisibleForTesting static final class QNameAndInterval {
+        private final byte[] qName;
+        private final int hashVal;
+        private final int intervalId;
+
+        QNameAndInterval( final String qName, final int intervalId ) {
+            this.qName = qName.getBytes();
+            this.hashVal = qName.hashCode();
+            this.intervalId = intervalId;
+        }
+
+        private QNameAndInterval( final Kryo kryo, final Input input ) {
+            final int nameLen = input.readInt();
+            qName = input.readBytes(nameLen);
+            hashVal = input.readInt();
+            intervalId = input.readInt();
+        }
+
+        private void serialize( final Kryo kryo, final Output output ) {
+            output.writeInt(qName.length);
+            output.writeBytes(qName);
+            output.writeInt(hashVal);
+            output.writeInt(intervalId);
+        }
+
+        public String getQName() { return new String(qName); }
+        public int getIntervalId() { return intervalId; }
+
+        @Override
+        public int hashCode() { return hashVal; }
+
+        @Override
+        public boolean equals( final Object obj ) {
+            return obj instanceof QNameAndInterval && equals((QNameAndInterval) obj);
+        }
+
+        public boolean equals( final QNameAndInterval that ) {
+            return this.intervalId == that.intervalId && Arrays.equals(this.qName, that.qName);
+        }
+
+        public boolean sameName( final byte[] name ) { return Arrays.equals(qName, name); }
+
+        public String toString() { return new String(qName)+" "+intervalId; }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<QNameAndInterval> {
+            @Override
+            public void write( final Kryo kryo, final Output output, final QNameAndInterval qNameAndInterval ) {
+                qNameAndInterval.serialize(kryo, output);
+            }
+
+            @Override
+            public QNameAndInterval read( final Kryo kryo, final Input input, final Class<QNameAndInterval> klass ) {
+                return new QNameAndInterval(kryo, input);
+            }
+        }
+    }
+
+    /**
+     * Class to find the template names associated with reads in specified intervals.
+     */
+    private static final class QNameFinder implements Function<GATKRead, Iterator<QNameAndInterval>> {
+        private final int maxQnamesPerInterval;
+        private final ReadMetadata metadata;
+        private final List<Interval> intervals;
+        private final int[] qNamesPerInterval;
+        private final Iterator<QNameAndInterval> noName = Collections.emptyIterator();
+        private int intervalsIndex = 0;
+
+        QNameFinder( final int maxQnamesPerInterval, final ReadMetadata metadata, final List<Interval> intervals ) {
+            this.maxQnamesPerInterval = maxQnamesPerInterval;
```

Comment:
Wouldn't it be possible to precompute an individual maxQNames for each interval? You might save some time collecting reads you don't want.